changelog.d/4263.bugfix

```diff
@@ -0,0 +1 @@
+Prevent crash on pagination.
```

Comment:
@erikjohnston please can we not use the word "crash" when we mean "throw an exception". "crash" sounds like the whole process dies.